File: 14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#

from typing import List

# @lc code=start


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 0:
            return ""
        ans = ""
        for i in range(len(strs[0])):
            curr = strs[0][i]
            for str in strs:
                if i >= len(str) or str[i] != curr:
                    return ans
            ans += curr
        return ans


# @lc code=end

File: 14.longest-common-prefix.cpp
/*
 * @lc app=leetcode id=14 lang=cpp
 *
 * [14] Longest Common Prefix
 */
#include <string>
#include <vector>
using namespace std;

// @lc code=start
class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        if (strs.size() == 0)
        {
            return "";
        }
        string ans;
        for (int i = 0; i < strs[0].size(); i++)
        {
            char curr = strs[0][i];
            for (const auto &str : strs)
            {
                if (i >= str.size() || str[i] != curr)
                {
                    return ans;
                }
            }
            ans += curr;
        }
        return ans;
    }
};
// @lc code=end
